curriculum/challenges/english/01-responsive-web-design/css-grid-magazine/part-051.md

```diff
@@ -0,0 +1,324 @@
+---
+id: 6148c434bd731d45617a76c6
+title: Part 51
+challengeType: 0
+dashedName: part-51
+---
+
+# --description--
+
+If you wanted to add more social icons, but keep them on the same row, you would need to update the `grid-template-columns` to create additional columns. As an alternative, you can use the `grid-auto-flow` property.
+
+This property takes either `row` or `column` as the first value, with an optional second value of `dense`. The `grid-auto-flow` uses an auto-placement algorithm to adjust the grid layout. Setting it to `column` will tell the algorithm to create new columns for content as needed. The `dense` value allows the algorithm to backtrack and fill holes in the grid with smaller items, which can result in items appearing out of order.
```

Comment:
```suggestion
This property takes either `row` or `column` as the first value, with an optional second value of `dense`. `grid-auto-flow` uses an auto-placement algorithm to adjust the grid layout. Setting it to `column` will tell the algorithm to create new columns for content as needed. The `dense` value allows the algorithm to backtrack and fill holes in the grid with smaller items, which can result in items appearing out of order.
```

Reply:
Also, is the `grid-auto-flow` property supposed to replace `grid-template-columns` here? Or just fall after the existing `grid-template-columns` property like in the next seed?:

```css
.social-icons {
  display: grid;
  font-size: 3rem;
  grid-template-columns: repeat(5, 1fr);
  grid-auto-flow: column;
}
```

Reply:
It falls after - the behaviour here being a minimum of five columns, adding more as needed.